database-plugins/src/main/java/co/cask/hydrator/plugin/DBUtils.java

```diff
@@ -58,6 +60,43 @@ public static void cleanup(Class<? extends Driver> driverClass) {
   }
 
   /**
+   * Ensures that the JDBC Driver specified in configuration is available and can be loaded. Also registers it with
+   * {@link DriverManager} if it is not already registered.
+   */
+  public static JDBCDriverShim ensureJDBCDriverIsAvailable(Class<? extends Driver> jdbcDriverClass,
+                                                           String connectionString,
+                                                           String jdbcPluginType, String jdbcPluginName)
+    throws IllegalAccessException, InstantiationException, SQLException {
+    try {
+      DriverManager.getDriver(connectionString);
+    } catch (SQLException e) {
+      // Driver not found. We will try to register it with the DriverManager.
+      LOG.debug("Plugin Type: {} and Plugin Name: {}; Driver Class: {} not found. Registering JDBC driver via shim {} ",
+                jdbcPluginType, jdbcPluginName, jdbcDriverClass.getName(),
+                JDBCDriverShim.class.getName());
+      JDBCDriverShim driverShim = new JDBCDriverShim(jdbcDriverClass.newInstance());
+      try {
+        DBUtils.deregisterAllDrivers(jdbcDriverClass);
+      } catch (NoSuchFieldException | ClassNotFoundException e1) {
+        LOG.error("Unable to deregister JDBC Driver class {}", jdbcDriverClass);
+      }
+      DriverManager.registerDriver(driverShim);
+      return driverShim;
+    }
+    return null;
+  }
+
+  public static void deregisterShim(JDBCDriverShim driverShim) {
```

Comment:
Seems like the `driverShim` can be null from the caller (in DBManager class above). So this param should have `@Nullable` and null should be handled in the method.